Tests/Unit/LoraKeysManagerFacade/SendCloudToDeviceMessageTest.cs

```diff
@@ -366,6 +385,47 @@ public async Task When_Querying_Devices_And_Finds_Single_Gateway_Class_C_Should_
             query.VerifyAll();
         }
 
+        [Fact]
+        public async Task When_Querying_Devices_And_Finds_No_Gateway_For_Class_C_Should_Return_InternalServerError()
+        {
+            const string devEUI = "0123456789";
+
+            var deviceTwin = new Twin
+            {
+                Properties = new TwinProperties()
+                {
+                    Desired = new TwinCollection($"{{\"DevAddr\": \"03010101\", \"ClassType\": \"C\"}}"),
```

Comment:
[nit] I would suggest to use the `DevAddr` type to ensure that we always have a valid `DevAddr` here (same for the `devEUI` on L391